logfire/_internal/config.py

```diff
@@ -797,6 +813,46 @@ def meter(self) -> metrics.Meter:
     def _initialize_credentials_from_token(self, token: str) -> LogfireCredentials | None:
         return LogfireCredentials.from_token(token, requests.Session(), self.base_url)
 
+    def _ensure_flush_after_lambda(self):
```

Comment:
How slow is this in the general case?

Is there an env var we can check before looping over all modules?

Reply:
```suggestion
    def _ensure_flush_after_aws_lambda(self):
```
Just to make it clear what we're talking about.

Reply:
> How slow is this in the general case?

With this test code:

```python
import time
import pandas
import sqlparse
import sqlalchemy
import sqlmodel
import sqlite3
import httpx
import django
import flask
import fastapi
import fastapi_cli
import uvicorn
import pydantic
import aiohttp
import pymongo
import redis

import logfire

start = time.time()
logfire.configure(
    token='foobar',
    # This speeds up `configure` by about 50% since it means it doesn't check git
    service_version='1.0.0',
)
end = time.time()
print(f'{end - start}')
```

Plus some tweaks in the new method, it prints:

```
len(list(sys.modules.values()))=1820
ensure_flush_after_lambda took 0.0005262080812826753s
0.025816917419433594
```

So this adds about 2% to the time that `configure` takes when a large number of modules have been imported already, something which will usually take even longer.

Reply:
> Is there an env var we can check before looping over all modules?

Right now, probably. It's very hard to trust that AWS will reliably continue to set these variables in the future.

https://github.com/search?q=repo%3Aaws%2Faws-lambda-python-runtime-interface-client%20AWS_LAMBDA_FUNCTION_NAME&type=code indicates that the lambda runtime client doesn't assume the variables are always there. And indeed they might not be if someone is running the client themselves in an emulator or something, which they encourage.